lib/webconn.js

```diff
@@ -147,6 +142,51 @@ WebConn.prototype.httpRequest = function (pieceIndex, offset, length, cb) {
           cb(null, ret)
         }
       }
+    }
+    get.concat(opts, function (err, res, data) {
+      if (hasError) return
+      if (err) {
+        // Browsers allow HTTP redirects for simple cross-origin
+        // requests but not for requests that require preflight.
+        // Use a simple request to get the final URL after redirection
+        // and retry the original request with this URL if they're
+        // different.
+        //
+        // This same-origin test is imperfect, but it's simple,
+        // good for common cases, and is only ever falsely negative,
+        // never falsely positive.
+        //
+        // We can't use simple-get here because http-browserify doesn't
+        // expose the final URL.
```

Comment:
I came back to this, and I now think we actually can use `simple-get` here. `browserify` replaced `http-browserify` with `stream-http` (see the list of shims [here](https://github.com/substack/node-browserify#compatibility)), which _does_ expose `res.url` (see https://github.com/jhiesey/stream-http/pull/40).

Here's a proof of concept that you can browserify and run in the console at https://example.com:

```js
var get = require('simple-get')

function getPreflightRedirect(opts, cb) {
  return get(opts, function (err, res, data) {
    if (opts.url.startsWith(window.location.origin + '/')) {
      return cb(err)
    }

    return get({ url: opts.url, method: 'HEAD' }, function (err, res, data) {
      if (err) return cb(err)
      if (res.url === opts.url) return cb(Exception('url did not redirect'))

      opts.url = res.url
      return get(opts, cb)
    })
  })
}

var opts = {
  url: 'https://httpbin.org/redirect-to?url=https%3A%2F%2Fhttpbin.org%2F',
  headers: {
    Range: 'bytes=0-'
  }
}
getPreflightRedirect(opts, function (err, res, data) {
  if (err) {
    console.error(err)
  }
  else {
    console.log("final url: " + res.url)
  }
})
```